monoloco/visuals/printer.py

```diff
@@ -59,21 +60,25 @@ def __init__(self, image, output_path, kk, args):
         self.kk = kk
         self.output_types = args.output_types
         self.z_max = args.z_max  # set max distance to show instances
-        self.show = args.show
-        self.show_all = args.show_all
-        self.save = not args.no_save
+        self.show_all = args.show_all or args.webcam
+        self.show = args.show_all or args.webcam
+        self.save = not args.no_save and not args.webcam
+        self.plt_close = not args.webcam
+        self.args = args
```

Comment:
I think you are only using activities, as extra arg, can you just add self.activities=args.activities?